masonite/view.py

```diff
@@ -12,27 +15,129 @@ def view(template='index', dictionary={}):
 
 
 class View(object):
+    """
+    Render template view
+    """
 
     def __init__(self):
         self.dictionary = {}
         self.composers = {}
+        self.container = None
+        self.cache = None
+        self.cache_time = None
+        self.cache_type = None
+        self.file_cache = None
         self.env = Environment(
             loader=PackageLoader('resources', 'templates'),
             autoescape=select_autoescape(['html', 'xml'])
         )
 
+    def load_container(self, container):
+        self.container = container
+        return self
+
+    def cache_for(self, time=None, type=None):
+        """
+        Set time and type for cache
+        """
+
+        self.cache = True
+        self.cache_time = time
+        self.cache_type = type
+        return self
+
+    def __create_cache_template(self, template):
+        """
+        Save in the cache the template
+        """
+
+        template = template + ":" + str(time.time()) + '.html'
+        self.file_cache = self.container.make('Cache').store(
+            template, self.rendered_template
+        )
+
+    def __cached_template_exists(self, template):
+        """
+        Check if template is cached
+        """
+
+        if not self.file_cache:
+            return False
+
+        cache_config = self.container.make('CacheConfig')
+        path = cache_config.DRIVERS['disk']['location'] + "/" + self.file_cache
+        if os.path.exists(path):
+            return True
+        else:
+            return False
+
+    def __expired_cache(self):
+        """
+        Check if cache is expired
+        """
+
+        # It's the first time
+        if not self.file_cache:
```

Comment:
This `self.file_cache` needs to change. This doesn't work if the view is only rendered once and only works because the view renders 2 times in the test. For example do this in test:

```python
    # The first render normally
    # view('test').cache_for(5, 'minutes')

    # Second time render from cache
    assert view('test').cache_for(5, 'minutes').rendered_template == 'test'
```

comment out the first one and the template doesn't cache anymore. The expired cache needs to look inside the cache directory for the cached template and not an attribute on the class